qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -0,0 +1,178 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class defining the main chain of a peptide."""
+from typing import List, Sequence
+
+from ...exceptions.invalid_side_chain_exception import (
+    InvalidSideChainException,
+)
+from ...exceptions.invalid_size_exception import InvalidSizeException
+from ..beads.main_bead import MainBead
+from .base_chain import BaseChain
+from .side_chain import SideChain
+
+
+class MainChain(BaseChain):
+    """A class defining a main chain of a peptide."""
+
+    def __init__(
+        self,
+        main_chain_len: int,
+        main_chain_residue_sequences: List[str],
+        side_chain_lens: List[int],
+        side_chain_residue_sequences: List[str],
+    ):
+        """
+        Args:
+            main_chain_len: Length of the main chain of a peptide.
+            main_chain_residue_sequences: List of characters that define residues for the main chain.
+            side_chain_lens: List of lengths of all side chains.
+            side_chain_residue_sequences: List of characters that define residues for all side
+            beads.
```

Comment:
The `_build_main_chain` calls validation methods which can raise errors. These need to be documented here.

Reply:
Thanks